src/js/components/AppListFilterComponent.jsx

```diff
@@ -17,24 +20,78 @@ var AppListFilterComponent = React.createClass({
     };
   },
 
-  updateFilterText: function (event) {
+  componentDidMount: function () {
+    this.updateFilterText();
+  },
+
+  componentWillReceiveProps: function () {
+    this.updateFilterText();
+  },
+
+  setQueryParam: function (filterText) {
+    var router = this.context.router;
+
+    var queryParams = router.getCurrentQuery();
+
+    if (filterText != null && filterText !== "") {
+      Object.assign(queryParams, {
+        filterText: encodeURIComponent(filterText)
```

Comment:
fair enough. I should stop reading javascript performance articles before sleep... :laughing: